pkg/controller/cassandrabackup/backup.go

```diff
@@ -0,0 +1,93 @@
+package cassandrabackup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/Orange-OpenSource/casskop/pkg/apis/db/v1alpha1"
+	"github.com/Orange-OpenSource/casskop/pkg/backrest"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupClient struct {
+	backup *api.CassandraBackup
+	client client.Client
+}
+
+func backup(
+	backrestClient *backrest.Client,
+	backupClient *backupClient,
+	logging *logrus.Entry,
+	recorder record.EventRecorder) {
+
+	operationID, err := backrestClient.PerformBackup(backupClient.backup)
+
+	if err != nil {
+		logging.Error(err, fmt.Sprintf("Error while starting backup operation"))
+		recorder.Event(backupClient.backup,
+			corev1.EventTypeNormal,
+			"BackupNotInitiated",
+			fmt.Sprintf("Backup of datacenter %s of cluster %s to %s under snapshot %s failed.",
+				backupClient.backup.Spec.Datacenter, backupClient.backup.Spec.CassandraCluster,
+				backupClient.backup.Spec.StorageLocation, backupClient.backup.Spec.SnapshotTag))
+		return
+	}
+
+	recorder.Event(backupClient.backup,
+		corev1.EventTypeNormal,
+		"BackupInitiated",
+		fmt.Sprintf("Task initiated to backup datacenter %s of cluster %s to %s under snapshot %s",
+			backupClient.backup.Spec.Datacenter, backupClient.backup.Spec.CassandraCluster,
+			backupClient.backup.Spec.StorageLocation, backupClient.backup.Spec.SnapshotTag))
+
+	for range time.NewTicker(2 * time.Second).C {
+		if status, err := backrestClient.BackupStatusByID(operationID); err != nil {
+			logging.Error(err, fmt.Sprintf("Error while finding submitted backup operation %v", operationID))
+			break
+		} else {
+			backupClient.updateStatus(status, logging)
```

Comment:
You might be in a case that the Backup update failed, and you only log the error, but the object will not be updated :/
Shouldn't we catch the error and try again on the next iteration ?